utils/dataset_validator.py

```diff
@@ -0,0 +1,109 @@
+from flask_uuid import UUID_RE
+
+DATASET_NAME_LEN_MIN = 1
+DATASET_NAME_LEN_MAX = 100
+
+MIN_CLASSES = 2
+CLASS_NAME_LEN_MIN = 1
+CLASS_NAME_LEN_MAX = 100
+
+MIN_RECORDINGS_IN_CLASS = 2
+
+
+def validate(dataset, complete=False):
```

Comment:
this does two types of validation - that the structure of the dataset objects is correct, and that the contents of the dataset follow a set of rules. Do you think it makes sense to combine them. My thought is that since dataset objects are always created by the app, this is less needed.